.github/workflows/master-build.yml

```diff
@@ -0,0 +1,72 @@
+name: Master Build
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    name: Build Java ${{ matrix.java }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+        java:
+          - 8
+          - 11
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v1
+      - name: Setup java
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+      - name: Cache Gradle Modules
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/caches
+          key: gradle-caches-${{ hashFiles('**/*.gradle.kts') }}
+      - name: Cache Gradle Wrapper
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle/wrapper
+          key: gradle-wrapper-${{ hashFiles('gradle/wrapper/gradle-wrapper.properties') }}
+      - name: Execute Gradle build
+        run: ./gradlew build --stacktrace
+        shell: bash
+        env:
+          CI: true
+  publish:
+    name: Publish snapshots
```

Comment:
So this happens fully automatically after each release PR is merged, then how do we do non-snapshot releases?

Reply:
I need to add release-build.yml in a separate PR :)